File: backend/core/src/main/java/jct/pillorganizer/core/dto/DeviceAccessDto.java
package jct.pillorganizer.core.dto;

import io.micronaut.serde.annotation.Serdeable;
import jakarta.annotation.Nullable;

@Serdeable
public record DeviceAccessDto(
        String deviceId,
        @Nullable String claimId,
        String nickname,
        @Nullable String serialNo,
        @Nullable String modelId,
        String tenantId,
        String apiBase,
        boolean primaryUser,
        @Nullable String thingName,
        String tenantName,
        @Nullable Boolean notifications
) {
}

File: app/lib/apiv2/models/device_access_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'device_access_dto.g.dart';

@JsonSerializable()
class DeviceAccessDto {
  final String id;
  final String deviceId;
  final String nickname;
  final String serialNo;
  final String modelId;
  final String tenantId;
  final String apiBase;
  final bool primaryUser;
  final String thingName;

  DeviceAccessDto({
    required this.id,
    required this.deviceId,
    required this.nickname,
    required this.serialNo,
    required this.modelId,
    required this.tenantId,
    required this.apiBase,
    required this.primaryUser,
    required this.thingName
  });

  factory DeviceAccessDto.fromJson(Map<String, dynamic> json) =>
      _$DeviceAccessDtoFromJson(json);

  Map<String, dynamic> toJson() => _$DeviceAccessDtoToJson(this);
}
